app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,82 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if !Helpers.logged_in?(session)
+      redirect "/login"
+    else
+      @user = Helpers.current_user(session)
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    end
+  end
+
+  get '/tweets/new' do
+    if Helpers.logged_in?(session)
+      erb :'/tweets/new'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.new(content: params[:content])
+    @tweet.user_id = session[:user_id]
+
+    if @tweet.save
+      redirect "/tweets/#{@tweet.id}"
+    else
+      redirect "/tweets/new"
+    end
+  end
+
+  get '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+    if Helpers.logged_in?(session)
+      erb :'/tweets/show'
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets/:id/edit' do 
+    if Helpers.logged_in?(session)
+      @tweet = Tweet.find(params[:id])
+        if @tweet.user_id == session[:user_id]
```

Comment:
I don't see a lot of students checking the tweet's user id. Good job!

Also, you can take advantage of the current_user method instead of using the session
`if @tweet.user_id == Helpers.current_user(session).id`